server.js

```diff
@@ -3474,17 +3476,23 @@ function socketServer() {
             }
 
             if (doesObjectExist(msgContent.object)) {
-                if (!realityEditorSocketArray[socket.id]) realityEditorSocketArray[socket.id] = [];
+                if (!realityEditorSocketSubscriptions.some(s => s.socket === socket)) {
+                    const subscriptions = [];
+                    subscriptions.socket = socket;
+                    realityEditorSocketSubscriptions.push(subscriptions);
```

Comment:
Little bit confused about adding a socket property to an array, rather than using an object, is there a reason to structure it like this? (I'm not always up-to-date with clean coding conventions so I figure you have a valid reason, just curious)

Reply:
Looking at it now, I cannot come up with a good answer for this. I'll push a commit with a more sensible arrangement!